.github/workflows/index-api.yaml

```
@@ -0,0 +1,29 @@
+name: Index API docs and upload to Algolia
+
+on:
+  schedule:
+    - cron: '0 0 * * 0'  # Runs every Sunday at midnight
+  workflow_dispatch:
+
+jobs:
+  update-algolia-index:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+
+    - name: Set up Node.js
+      uses: actions/setup-node@v1
```

This setup is currently on v3. Not sure the difference tho

https://github.com/actions/setup-node#usage
```suggestion
      uses: actions/setup-node@v3
```